ConsoleApp/ConsoleApp/Program.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+
+namespace ConsoleApp
+{
+    //i.safontev/homework/14-generics
+    public class Stack<T>
+    {
+        public T[] Arr { get; set; }
+        public int Top { get; set; }
+        public int Count { get; set; }
+
+        public Stack(int Count)
+        {
+            this.Count = Count;
+            Arr = new T[Count];
+        }
+
+        public void Push(T Element) 
+        {
+            int k = 1;
+            if (Top == Count)
+            {
+                T[] newArray = new T[Top + 1];
+                Array.Copy(Arr, 0, newArray, 0, Top);
+                Arr = newArray;
+                k = 0;
+            }
+            Arr[Top] = Element;
+            Top += k;
+        }
+
+        public T Pop()
+        {
+            if (Top > 0)
+            {
+                Console.WriteLine(Top);
+                T lastElement = Arr[Top];
+                Top--;
+                return lastElement;
+            }
+            return default(T);
```

Comment:
in case if there are no elements it's better to throw exception because you can have element in stack equal to default value